internal/configconverter/remove_memory_ballast_key.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configconverter
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+// RemoveMemoryBallastKey is a MapConverter that removes a memory_ballast on a
+// extension config if it exists.
+type RemoveMemoryBallastKey struct{}
+
+func removeMemoryBallast(strList []interface{}) []interface{} {
+	ret := make([]interface{}, 0)
+	for i, v := range strList {
+		if v == "memory_ballast" {
+			ret = append(ret, strList[:i]...)
+			return append(ret, strList[i+1:]...)
+		}
+	}
```

Comment:
Seems like an extensions slice with no `memory_ballast` will be overridden to an empty slice. Please add a test case for that even if I'm wrong

Reply:
I think you're correct about the method's inconsistency even though we are making sure to send slices with "memory_ballast" string. None the less, I have corrected the bug and added two new test cases. A test case just to test the helper function and an actual converter test without memory_ballast in the config